### PR TITLE
UI-8 Dropdown

### PR DESCRIPTION
**Pull request by lmiller1990**
image

### CODE REVIEW

File: src/Dropdown/Dropdown.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+)
+
+
+const searchIcon = (
+  <FontAwesomeIcon
+    icon={faSearch}
+  />
+)
+
+const noop = () => {/***/}
```
Comment (MichaelDorkhom) [inline src/Dropdown/Dropdown.stories.tsx]: Can use lodash for this
https://lodash.com/docs/4.17.15#noop

Comment (lmiller1990) [reply src/Dropdown/Dropdown.stories.tsx]: ok

File: src/Dropdown/index.tsx [hunk 2]
```diff
@@ -0,0 +1,136 @@
+import React, { useState, useRef, useEffect } from 'react'
+import { InputGroup, FormControl, FormControlProps, } from 'react-bootstrap'
+
+import { Button } from '../Button'
+import './index.scss'
+import { ReplaceProps, BsPrefixProps } from 'react-bootstrap/helpers'
```
Comment (MichaelDorkhom) [inline src/Dropdown/index.tsx]: Can you move this to the first block of imports

File: src/Dropdown/index.tsx [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+
+interface IProps {
+  data?: string[]
+  title: string
```
Comment (MichaelDorkhom) [inline src/Dropdown/index.tsx]: Should `title` be `placeholder` instead?

File: src/Dropdown/index.tsx [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+}
+
+const Dropdown: React.FC<IProps> = ({ children, data, title, searchIcon, onSelect, showSearchThreshold = 10 }) => {
+  const [showContent, setShowContent] = useState(false)
+  const [searchText, setSearchText] = useState('')
+  const [selectedItem, setSelectedItem] = useState('')
+  const node = useRef<HTMLDivElement>(null)
+
+  useEffect(() => {
+    document.addEventListener("mousedown", (e: MouseEvent) => {
```
Comment (MichaelDorkhom) [inline src/Dropdown/index.tsx]: Single quotes

File: src/Dropdown/index.tsx [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+          )
+        }
+      </ul>
+    </>
+  )
+
+  let filteredData = data;
+
+  if (data && searchText) {
+    filteredData = data.filter(x =>  x.toLowerCase().includes(searchText))
```
Comment (MichaelDorkhom) [inline src/Dropdown/index.tsx]: `searchText.toLowerCase()` if using `x.toLowerCase()`?

File: src/style/colors.scss [hunk 6]
```diff
@@ -16,3 +16,6 @@ $bgwhite: rgb(246, 247, 251);
 $validGreen: rgb(40, 167, 69);
 $invalidRed: rgb(220, 53, 69);
 $shadow: rgba(180, 180, 180, 0.3);
+
+$navy: #09325f;
+$dropdown-blue: #c5daf6;
```
Comment (MichaelDorkhom) [inline src/style/colors.scss]: Next line

File: src/Dropdown/index.scss [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+}
+
+.dropdown-wrapper {
+  position: relative;
+  width: $width;
+}
+
+.dropdown-closed {
+  padding: 5px $side 5px $side + 2;
+  background: #fcfcfc;
```
Comment (MichaelDorkhom) [inline src/Dropdown/index.scss]: `#fcfcfc` should probably be a constant, used multiple times here

File: src/Dropdown/index.tsx (hunk 3 above)
Comment (MichaelDorkhom) [inline src/Dropdown/index.tsx]: Should this be called `placeholder` instead of title?

File: src/Dropdown/Dropdown.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+stories.add('Components.Dropdown', () => {
+  return (
+    <>
+      <h3 className='m-5'>With List of Items + No Search</h3>
+      <div className='m-5'>
+        <Dropdown 
+          title='Select Hospital'
+          data={['Hospital A', 'Hospital B', 'Hospital C']}
+          onSelect={noop}
+          searchIcon={searchIcon}
```
Comment (MichaelDorkhom) [inline src/Dropdown/Dropdown.stories.tsx]: Don't think this needs to be a property, probably want consistency for the search icon

File: src/Dropdown/index.tsx [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+            </Button>
+          </InputGroup.Append>
+        )
+      }
+    </InputGroup>
+  )
+
+  const listItems = (items: string[], callback: (item: string) => void) => (
+    <>
+      <div className='dropdown-title'>{title}</div>
```
Comment (MichaelDorkhom) [inline src/Dropdown/index.tsx]: Do we need this? The form label should be separate from the dropdown

File: src/Dropdown/Dropdown.stories.tsx (hunk 8 above)
Comment (lmiller1990) [reply src/Dropdown/Dropdown.stories.tsx]: Yeah I guess, I didn't really want to bundle FontAwesome with the UI lib... I guess we can

File: src/Dropdown/index.tsx (hunk 5 above)
Comment (lmiller1990) [reply src/Dropdown/index.tsx]: we downcase the searchText when we update the state, so no need!

File: src/Dropdown/index.tsx (hunk 3 above)
Comment (lmiller1990) [reply src/Dropdown/index.tsx]: ok suyre

File: src/Dropdown/index.tsx (hunk 9 above)
Comment (lmiller1990) [reply src/Dropdown/index.tsx]: not quite sure what you mean by this, please explain more?

File: src/Dropdown/index.tsx (hunk 4 above)
Comment (lmiller1990) [reply src/Dropdown/index.tsx]: ok

File: src/style/colors.scss (hunk 6 above)
Comment (lmiller1990) [reply src/style/colors.scss]: yep

File: src/Dropdown/index.scss (hunk 7 above)
Comment (lmiller1990) [reply src/Dropdown/index.scss]: done

File: src/Dropdown/Dropdown.stories.tsx (hunk 8 above)
Comment (lmiller1990) [reply src/Dropdown/Dropdown.stories.tsx]: hmm can do, was trying to avoid the Font Awesome dependency

**Pull request merged**